assessment.sql

```diff
@@ -6,23 +6,52 @@ CREATE DATABASE "sql-crud-assessment";
 -- Question 1. Create a table called `licensees` to hold data in columns surname,given_name,gender,height,weight,dob,eye_color,hair_color.  See `data/licensees.csv` for example data.  Don't forget that all tables should have an id column as a unique identifier for a row.
 
 -- your code here
+CREATE TABLE licensees(
+  id SERIAL PRIMARY KEY,
+  surname CHARACTER VARYING,
+  given_name CHARACTER VARYING,
+  gender CHARACTER VARYING (1),
+  height INTEGER,
+  weight INTEGER,
+  dob DATE,
+  eye_color CHARACTER VARYING,
+  hair_color CHARACTER VARYING
+);
 
 -- Question 2. Add `Bishop, Caren, f, 63, 132, 1943-09-26, Brown, Black` into the table of licensees.
 
 -- your code here
+INSERT INTO licensees(surname,given_name,gender,height,weight,dob,eye_color,hair_color)
+VALUES('Bishop', 'Caren', 'f', 63, 132, '1943-09-26', 'Brown', 'Black')
+;
 
 -- Question 3. Load the data from `data/licensees.csv` into the `licensees` table.
 
 -- your code here
+\copy licensees(ssurname,given_name,gender,height,weight,dob,eye_color,hair_color) FROM 'data/licensees.csv' WITH (HEADER true, FORMAT csv)
 
 -- Question 4. Write a query to get all attributes of licensees with Hazel eye color and Brown or Black hair color.
 
 -- your code here
+SELECT COUNT (*)
+FROM licensees
+WHERE eye_color = 'Hazel' AND hair_color = 'Brown' OR 'Black'
+;
```

Comment:
This where clause is broken after the `OR`: `OR` expects a boolean expression on each side of it. Comparing the value in a column with a value literal yields a boolean; a string literal, on the other hand, is not a boolean.

Even after fixing that, the WHERE clause is incorrect in a subtle way. `AND` has precedence over `OR`, so it looks for either both eye color being hazel and hair color being brown, or just hair color being brown. You must use parentheses to enforce an alternative order of operations.

Finally, this does not retrieve any rows at all. It gets the number of rows that would have matched your query. 3/5